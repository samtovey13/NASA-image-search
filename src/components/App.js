import React from 'react';
import '../styles/App.css';
import Search from './Search';

function App() {
  return (
    <div className="App">
      <img 
        className="logo" 
        alt="NASA logo"
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg" 
        data-testid="NASA-logo"
      />
      <Search />
    </div>
  );
}

export default App;
