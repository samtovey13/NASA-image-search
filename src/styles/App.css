.App {
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-evenly;
  align-items: center;
  box-sizing: border-box;
}

.logo {
  margin: 10px;
  width: 472px;
  max-width: 100%;
}