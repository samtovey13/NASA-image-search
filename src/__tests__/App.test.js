import React from 'react';
import { render } from '@testing-library/react';
import App from '../components/App';

describe ("App", () => {
  it ("renders correctly", () => {
    const { asFragment } = render( <App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it ("renders the correct components", () => {
    const { getByTestId } = render(<App />);

    expect(getByTestId("NASA-logo")).toHaveClass("logo");
    expect(getByTestId("search-input")).toBeInTheDocument();
  });
});