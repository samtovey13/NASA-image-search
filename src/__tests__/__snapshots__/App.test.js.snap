// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="NASA logo"
      class="logo"
      data-testid="NASA-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <form
      class="search-form"
    >
      <input
        class="search-input"
        data-testid="search-input"
        placeholder="Search"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        GO
      </button>
    </form>
  </div>
</DocumentFragment>
`;
